### PR TITLE
Soar 9910 ExtractIt 3 Issues Fix

### PR DESCRIPTION
**Pull request by aholland-r7**
### Description

Remove parenthesis at end of extracted URL if not supposed to be there.
Fix issue with extracting dates from .xlsx file
Updated error message in extractor.py
Updated help.md examples

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/extractit/help.md [hunk 1]
```diff
@@ -43,9 +43,9 @@ This action extracts all UUIDs, CVEs, dates, domains, emails, filepaths, IOCs, I
...
+|date_format|string|None|True|Dates matching this format are extracted - If All Formats is selected, found dates will be processed in the listed order documented|['dd/mm/yyyy', 'mm/dd/yyyy', 'dd\mm\yyyy', 'dd.mm.yyyy', 'dd-mm-yyyy', 'dd.mm.yy', 'dd-mm-yy', 'dd/mm/yy', 'dd\mm\yy', 'mm\dd\yyyy', 'mm.dd.yyyy', 'mm-dd-yyyy', 'mm/dd/yy', 'mm\dd\yy', 'mm.dd.yy', 'mm-dd-yy', 'dd/mmm/yyyy', 'dd\mmm\yyyy', 'dd.mmm.yyyy', 'dd-mmm-yyyy', 'dd/mmm/yy', 'dd\mmm\yy', 'dd.mmm.yy', 'dd-mmm-yy', 'yyyy.mm.dd', 'yyyy-mm-dd', 'yyyy/mm/dd', 'yyyy\mm\dd', 'yyyy.mmm.dd', 'yyyy-mmm-dd', 'yyyy/mmm/dd', 'yyyy\mmm\dd', 'yy.mm.dd', 'yy-mm-dd', 'yy/mm/dd', 'yy\mm\dd', 'yyyy-mm-ddThh:mm', 'yyyy-mm-ddThh:mm:ss', 'All Formats', '']|dd/mm/yyyy|
```
Comment (igorski-r7) [inline plugins/extractit/help.md:46]: I am not sure if that will be working. Icon validators fail due to that change 😞

File: plugins/extractit/help.md [hunk 2]
```diff
@@ -266,7 +266,7 @@ This action extracts all email addresses from a string or file.
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
 |file|bytes|None|False|Input file as bytes, supports text and binary file types such as PDF, DOCX, XLSX, PPTX, ODT, ODP, ODS|None|dXNlckBleGFtcGxlLmNvbSBpcyBhbiBleGFtcGxlIGVtYWls|
-|str|string|None|False|Input string|None|user@example.com is an example email|
+|str|string|None|False|Input string|None|https://example.com is an example email|
```
Comment (igorski-r7) [inline plugins/extractit/help.md:269]: Should that one be changed?

File: plugins/extractit/help.md [hunk 3]
```diff
@@ -302,7 +302,7 @@ This action extracts all file paths from a string or file.
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
 |file|bytes|None|False|Input file as bytes, supports text and binary file types such as PDF, DOCX, XLSX, PPTX, ODT, ODP, ODS|None|L3RtcC9pbWFnZS5qcGcgaXMgYW4gZXhhbXBsZSBmaWxlIHBhdGg=|
-|str|string|None|False|Input string|None|/tmp/image.jpg is an example file path|
+|str|string|None|False|Input string|None|/tmp/https://example.com is an example file path|
```
Comment (igorski-r7) [inline plugins/extractit/help.md:305]: Same as above. Shouldn't that be a path?

File: plugins/extractit/help.md [hunk 4]
```diff
@@ -338,7 +338,7 @@ This action extracts all Indicators of Compromise from a string or file.
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
 |file|bytes|None|False|Input file as bytes, supports text and binary file types such as PDF, DOCX, XLSX, PPTX, ODT, ODP, ODS|None|dXNlckBleGFtcGxlLmNvbSwgMTk4LjUxLjEwMC4xMDAgYW5kIDQ0ZDg4NjEyZmVhOGE4ZjM2ZGU4MmUxMjc4YWJiMDJmIGFyZSBJT0MgZXhhbXBsZXM=|
-|str|string|None|False|Input string|None|user@example.com, 198.51.100.100 and 44d88612fea8a8f36de82e1278abb02f are IOC examples|
+|str|string|None|False|Input string|None|https://example.com, https://example.com and 9de5069c5afe602b2ea0a04b66beb2c0 are IOC examples|
```
Comment (igorski-r7) [inline plugins/extractit/help.md:341]:
```suggestion
|str|string|None|False|Input string|None|user@example.com, 198.51.100.100 and 44d88612fea8a8f36de82e1278abb02f are IOC examples|
```

File: plugins/extractit/help.md [hunk 5]
```diff
@@ -377,7 +377,7 @@ This action extracts all IPv4 and IPv6 addresses from a string or file.
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
 |file|bytes|None|False|Input file as bytes, supports text and binary file types such as PDF, DOCX, XLSX, PPTX, ODT, ODP, ODS|None|MTk4LjUxLjEwMC4xMDAgYW5kIDIwMDE6ZGI4Ojg6NDo6MiBhcmUgc2FtcGxlIElQIGFkZHJlc3Nlcw==|
-|str|string|None|False|Input string|None|198.51.100.100 and 2001:db8:8:4::2 are sample IP addresses|
+|str|string|None|False|Input string|None|https://example.com and 2001:db8:8:4::2 are sample IP addresses|
```
Comment (igorski-r7) [inline plugins/extractit/help.md:380]: Same for IP address. It seems that plugin regenerate cmd changed all the elements here 😄

File: plugins/extractit/icon_extractit/actions/url_extractor/action.py [hunk 6]
```diff
@@ -16,12 +25,40 @@ def __init__(self):
...
+        elif provided_file:
+            try:
+                if keep_original_url:
+                    provided_file = base64.b64decode(provided_file.encode(DEFAULT_ENCODING)).decode(DEFAULT_ENCODING)
+                    return provided_file
+                else:
+                    provided_file = urllib.parse.unquote(
+                        base64.b64decode(provided_file.encode(DEFAULT_ENCODING)).decode(DEFAULT_ENCODING)
+                    )
+                    return provided_file
```
Comment (igorski-r7) [inline plugins/extractit/icon_extractit/actions/url_extractor/action.py:61]:
```suggestion
                if keep_original_url:
                    provided_file = base64.b64decode(provided_file.encode(DEFAULT_ENCODING)).decode(DEFAULT_ENCODING)
                else:
                    provided_file = urllib.parse.unquote(
base64.b64decode(provided_file.encode(DEFAULT_ENCODING)).decode(DEFAULT_ENCODING)
                    )
                return provided_file0
```

**Review by igorski-r7 — COMMENTED** [review]
Looks great! Just left some small comments 😄

**Review by cmcnally-r7 — COMMENTED** [review]
Could you make this update 3.0.6, and just rebase once this PR is merged to develop?
https://github.com/rapid7/insightconnect-plugins/pull/1382

**Review by igorski-r7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**